cli/CommandExecutor.cpp

```diff
@@ -0,0 +1,200 @@
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "cli/CommandExecutor.hpp"
+
+#include <cstddef>
+#include <cstdio>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "catalog/CatalogAttribute.hpp"
+#include "catalog/CatalogDatabase.hpp"
+#include "catalog/CatalogRelation.hpp"
+#include "catalog/CatalogRelationSchema.hpp"
+#include "cli/PrintToScreen.hpp"
+#include "parser/ParseStatement.hpp"
+#include "utility/PtrVector.hpp"
+#include "utility/Macros.hpp"
+#include "utility/SqlError.hpp"
+
+#include "gflags/gflags.h"
+#include "glog/logging.h"
+
+using std::fprintf;
+using std::fputc;
+using std::fputs;
+using std::size_t;
+using std::vector;
+using std::string;
+
+namespace quickstep {
+
+void CommandExecutor::executeCommand(const ParseStatement &statement,
+                                     const CatalogDatabase *catalog_database,
+                                     FILE *out) {
+  const ParseCommand &command = static_cast<const ParseCommand &>(statement);
+  const PtrVector<ParseString> *arguments = command.arguments();
+  std::string command_str = command.command()->value();
+  if (command_str.compare("\\dt") == 0) {
+    executeDescribeDatabase(arguments, catalog_database, out);
+  } else if (command_str.compare("\\d") == 0) {
+    if (arguments->size() == 0) {
+      executeDescribeDatabase(arguments, catalog_database, out);
+    } else {
+      executeDescribeTable(arguments, catalog_database, out);
+    }
+  } else {
+    THROW_SQL_ERROR_AT(command.command()) << "Invalid Command";
+  }
+}
+
+void CommandExecutor::executeDescribeDatabase(
+    const PtrVector<ParseString> *arguments,
+    const CatalogDatabase *catalog_database, FILE *out) {
+  std::size_t relation_count = 0;
+  // Column width initialized to 6 to take into account the header name
+  // and the column value table
+  int max_column_width = 6;
+  const CatalogRelation *relation;
+  if (arguments->size() == 0) {
+    for (auto rel_it = catalog_database->begin();
+         rel_it != catalog_database->end(); ++rel_it) {
+      int column_width = static_cast<int>(rel_it->getName().length());
+      max_column_width =
+          max_column_width < column_width ? column_width : max_column_width;
+      relation_count++;
+    }
+  } else {
+    const ParseString &table_name = arguments->front();
+    const std::string &table_name_val = table_name.value();
+    relation = catalog_database->getRelationByName(table_name_val);
+    int column_width = static_cast<int>(relation->getName().length());
+    max_column_width =
+        max_column_width < column_width ? column_width : max_column_width;
+    relation_count++;
```

Comment:
It should always be `1`, right?

Also, never use `i++`, unless you need the value of `i` before `++`.